crates/solver/src/settlement_submission.rs

```diff
@@ -108,25 +108,26 @@ impl SolutionSubmitter {
             TransactionStrategy::Flashbots {
                 flashbots_api,
                 max_confirm_time,
+                retry_interval,
                 flashbots_tip,
             } => {
-                let submitter = FlashbotsSolutionSubmitter::new(
+                let submitter = Submitter::new(
                     &self.web3,
                     &self.contract,
                     &account,
                     flashbots_api,
                     self.gas_price_estimator.as_ref(),
-                    self.gas_price_cap,
                 )?;
-                let result = submitter
-                    .submit(
-                        self.target_confirm_time,
-                        SystemTime::now() + *max_confirm_time,
-                        settlement,
-                        gas_estimate,
-                        *flashbots_tip,
-                    )
-                    .await;
+                let params = SubmitterParams {
+                    target_confirm_time: self.target_confirm_time,
+                    gas_estimate,
+                    gas_price_cap: self.gas_price_cap,
+                    deadline: Some(Instant::now() + *max_confirm_time),
+                    pay_gas_to_coinbase: None,
```

Comment:
Instead of making this part of the general "Submitter" can't this be an implementation detail of `ArcherApi`, where the payment to `coinbase` gets added to the interactions before encoding the calldata?

Reply:
This would require the interface to take the MethodBuilder instead of already encoded calldata, which I'd also be in favour of.

Reply:
Removed as its not needed.